pkg/operator/ceph/disruption/clusterdisruption/reconcile.go

```diff
@@ -257,3 +253,41 @@ func (c *ClusterMap) GetClusterNamespaces() []string {
 	}
 	return namespaces
 }
+
+func (r *ReconcileClusterDisruption) deleteDrainCanaryPods() error {
+	canaryLabels := client.MatchingLabels{k8sutil.AppAttr: "rook-ceph-drain-canary"}
+	drainCanaryList := &appsv1.DeploymentList{}
+	err := r.client.List(context.TODO(), drainCanaryList, canaryLabels, client.InNamespace(r.context.OperatorNamespace))
```

Comment:
Can we `Delete()` with the label here also instead of `List()` then `Delete()` individually?

Reply:
using `DeleteAllOf`